src/common/evaluable.ts

```diff
@@ -60,10 +60,17 @@ export interface Evaluable {
    * Simplifies this Evaluable when possible.
    *
    * @param {Context} ctx context for the evaluation
-   * @param {string[]} ignoreKeys keys to be considered present even if their not present in the context
+   * @param {false | string[]} alwaysEvaluate keys to be considered present even if they are not present in the context
+   * @param {false | string[]} deferEvaluate keys to be considered not present unless they are also in `alwaysEvaluate`;
+   *  when `alwaysEvaluate` is set to `false` and `deferEvaluate` is an array, every key that is not in `deferEvaluate`
+   *  is considered to be present and thus will be evaluated
    * @returns {Result | Evaluable} simplified value or itself
    */
-  simplify(ctx: Context, ignoreKeys: string[]): Result | Evaluable
+  simplify(
+    ctx: Context,
+    alwaysEvaluate?: false | string[],
+    deferEvaluate?: false | string[]
```

Comment:
For consideration, https://github.com/briza-insurance/illogical/blob/e7405e310742e179ef0d4981669b4b4a55064e0b/src/operand/reference.ts#L30-L32 to extend here to support **regexp** (small bit form that experimental PR: https://github.com/briza-insurance/illogical/pull/194)

Reply:
Interesting. I think it would be backward compatible if in the future we add RegExp.

Reply:
Yeah, it should. Just for consideration here.